src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,116 @@
 /* eslint-disable */
 import React from 'react';
 
+import { useAppSelector } from '../../app/hooks';
+import { useDispatch } from 'react-redux';
+import { Todo } from '../../types/Todo';
+import { currentTodoSlice } from '../../features/currentTodo';
+import { Status } from '../../types/Status';
+
+import classNames from 'classnames';
+
 export const TodoList: React.FC = () => {
+  const todosFilter = useAppSelector(state => state.todos);
+  const filter = useAppSelector(state => state.filter);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useDispatch();
+  const handleSelectTodo = (todo: Todo) => {
+    dispatch(currentTodoSlice.actions.selectTodo(todo));
+  };
+  function getFilteredTodos(inheritTodods: Todo[]) {
+    let filteredTodos = [...inheritTodods];
+    const adaptedQuery = filter.query.trim().toLowerCase();
+    switch (filter.status) {
+      case Status.Active:
+        filteredTodos = inheritTodods.filter(todo => !todo.completed);
+        break;
+      case Status.Completed:
+        filteredTodos = inheritTodods.filter(todo => todo.completed);
+        break;
+      default:
+        break;
+    }
+    if (adaptedQuery) {
+      return filteredTodos.filter(todo =>
+        todo.title.toLowerCase().includes(adaptedQuery),
+      );
+    }
+    return filteredTodos;
+  }
```

Comment:
The `getFilteredTodos` function is inside the `TodoList` component, which means it will be recreated every time the component re-renders. This is not efficient. You could move this function outside of the component or use `useMemo` or `useCallback` to prevent unnecessary recreations of this function.